include/morphio/collection.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <morphio/morphology.h>
+#include <morphio/mut/morphology.h>
+
+namespace morphio {
+
+class CollectionImpl
+{
```

Comment:
can we just forward decl this, and move the `load`s to `collection.cpp`?  I *try* (often unsuccessfully) to keep the headers as simple as possible.

Reply:
Good point. Solved via explicit instantiation.